src/org/loklak/http/ClientConnection.java

```diff
@@ -363,8 +364,9 @@ public static String getRedirect(String urlstring, boolean useAuthentication, bo
                     if (debugLog) {
                         DAO.log("Fetched Meta redirect URL for " + urlstring + " : " + metaURL);
                     }
-                    if (!metaURL.startsWith("http") && !urlstring.endsWith("/")) {
-                        return urlstring + "/" + metaURL;
+                    if (!metaURL.startsWith("http")) {
+                        URL u = new URL(new URL(urlstring), metaURL);
+                        return u.toString();
```

Comment:
It is used to build a complete URL from two parts - base and relative.
For example, when the base URL is `http://yacy.net` and the relative URL is `en/index.html`, it would give `http://yacy.net/en/index.html`.

It will also take care of other stuff - 
* `http://yacy.net/` + `/en/index.html` = `http://yacy.net/en/index.html` (take care of slashes).
* `http://a.com/b/c` + `../d` = `http://a.com/b/d`.